lib/Runtime/Provisioner/Provisioner.cpp

```diff
@@ -29,85 +30,96 @@ Provisioner::Provisioner(DeviceManagerMapTy &devices) {
   for (auto &device : devices) {
     devices_.push_back(device.second.get());
   }
+  auto backendKind = devices[0]->getBackendKind();
+  backend_.reset(createBackend(backendKind));
 }
 
+void walkNetwork(DAGNode *node,
+                 std::map<DeviceIDTy, std::vector<DAGNode *>> &logicalDevices) {
```

Comment:
nit: is the consistent traversal order in logicalDevices necessary?  If not, maybe std::unordered_map is better :)

Reply:
I went with map because the number of elements is small enough it is more memory efficient than unordered_map.

Reply:
I see. I didn't notice the size of the map :) So map should be better!